composer/utils/checkpoint.py

```diff
@@ -318,7 +322,7 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
                     )
                     received_file_object = file_object[0]
                     assert received_file_object is not None
-                    if receiver and not os.path.exists(full_path):
+                    if receiver and not os.path.exists(full_path) and dist.get_local_rank() == 0:
```

Comment:
is `os.path.exists` not a sufficient check for multi GPU write? or would it race?

Reply:
Not 100% sure if it would race, but adding this 0 rank condition to be more safe.